client/src/components/Modals/EditTeam/index.tsx

```diff
@@ -0,0 +1,195 @@
+  ModalProps,
+} from 'semantic-ui-react';
+import Swal from 'sweetalert2';
+import { ITeam } from 'ssw-common';
+
+import './styles.scss';
+import { useTeams } from '../../../contexts';
+import { createManyTeams, updateManyTeams } from '../../../api';
+
+const TeamModalTrigger: FC<ButtonProps> = ({ ...rest }): ReactElement => (
+  <Button {...rest} size="tiny" circular icon style={{ background: 'none' }}>
+    <Icon name="pencil" />
+  </Button>
+);
+
+const TeamModal: FC<ModalProps> = ({ ...rest }): ReactElement => {
+  const { teams, fetchTeams } = useTeams();
+  const [isOpen, setIsOpen] = useState(false);
+  const [formValues, setFormValues] = useState<ITeam[]>([]);
+
+  const [clonedTeams, setClonedTeams] = useState<Partial<ITeam>[]>([]);
+
+  useEffect(() => {
+    setFormValues(teams);
+    const clone = teams.map((team) => ({
+      _id: team._id,
+      name: team.name,
+      color: team.color,
+      active: team.active,
+    }));
+
+    setClonedTeams([...clone]);
+    fetchTeams();
+
+    // DON'T DO THIS UNLESS AMIT SAYS SO
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
+
+  // Add a new input field in form
+  const addInputLine = (): void => {
+    setFormValues([
+      ...formValues,
+      { _id: 'NEW', name: '', color: '#3d4f91', active: true },
+    ]);
+  };
+
+  // Remove an input field in form
+  const removeInputLine = (i: number): void => {
+    const newFormValues = [...formValues];
+    newFormValues.splice(i, 1);
+    setFormValues(newFormValues);
+  };
+
+  // Update team names (input type = text)
+  const updateInputText = (i: number, text: string): void => {
+    const formValue = formValues[i];
+    formValue.name = text;
+    setFormValues([...formValues]);
+  };
+
+  // Update team colors (input type = color)
+  const updateInputColor = (i: number, color: string): void => {
+    const formValue = formValues[i];
+    formValue.color = color;
+    setFormValues([...formValues]);
+  };
+
+  // Check if team name inputs contain duplicates
+  const isDuplicate = (): boolean => {
+    const valuesName = formValues.map((item) => item.name);
+    return valuesName.some((item, idx) => valuesName.indexOf(item) !== idx);
+  };
+
+  // Save newTeams and changedTeams
+  const saveTeam = (): void => {
+    // Save newly created teams in newTeams
+    const newTeams = formValues.filter((team) => team._id === 'NEW');
+    // Take out _id = "NEW" field from newTeams
+    const parsedNewTeams: Partial<ITeam>[] = newTeams.map((team) => ({
+      name: team.name,
+      color: team.color,
+      active: team.active,
+    }));
+
+    // Save the changes made to existing team in changedTeams
+    const changedTeams: ITeam[] = [];
+
+    clonedTeams.forEach((team, index) => {
+      const diffName = team.name !== formValues[index].name;
+      const diffColor = team.color !== formValues[index].color;
+      if (diffName || diffColor) {
+        changedTeams.push(formValues[index]);
+      }
+    });
```

Comment:
Maybe create a helper function for the check here? Then you could just run `changedTeams = clonedTeams.filter(isUpdated)` or similar.